### PR TITLE
Deploy NSG into resource group of the existing vnet.

### PR DESCRIPTION
**Pull request by KimForss**
### Problem 

The current design will create NSGs in the same resource group as the VMs. When deploying multiple SDUs and having the VNet in another resource group the subnets will end up with multiple NSGs all with the same name (as each SDU deployment creates a new one)

### Solution

Add the NSGs in the same Resource group as the VNet.

### CODE REVIEW

File: deploy/terraform/modules/hdb_node/infrastructure.tf [hunk 1]
```diff
@@ -36,16 +36,16 @@ data "azurerm_subnet" "subnet-sap-db" {
 resource "azurerm_network_security_group" "nsg-admin" {
   count               = local.enable_deployment ? (local.sub_admin_nsg_exists ? 0 : 1) : 0
   name                = local.sub_admin_nsg_name
-  location            = var.resource-group[0].location
```
Comment (rukawata) [inline deploy/terraform/modules/hdb_node/infrastructure.tf:39]: location is missing:
```suggestion
  virtual_network_name = var.vnet-sap[0].location
```

File: deploy/terraform/modules/hdb_node/infrastructure.tf [hunk 2]
```diff
@@ -36,16 +36,16 @@ data "azurerm_subnet" "subnet-sap-db" {
...
-  resource_group_name = var.resource-group[0].name
+  resource_group_name  = var.vnet-sap[0].resource_group_name
+  virtual_network_name = var.vnet-sap[0].name
```
Comment (rukawata) [inline deploy/terraform/modules/hdb_node/infrastructure.tf:40]: virtual_network_name is not part of the nsg attribute.
You can use [azurerm_subnet_network_security_group_association](https://www.terraform.io/docs/providers/azurerm/r/subnet_network_security_group_association.html) to link the nsg and subnet.

Comment (KimForss) [reply deploy/terraform/modules/hdb_node/infrastructure.tf]: fixed

File: deploy/terraform/modules/hdb_node/infrastructure.tf (hunk 1 above)
Comment (KimForss) [reply deploy/terraform/modules/hdb_node/infrastructure.tf:39]: fixed

**Pull request merged**